src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs

```diff
@@ -276,14 +276,11 @@ static string EncodeIndices(IEnumerable<int> indices, int additionalLocationsLen
             return $"https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/{id.ToLowerInvariant()}";
         }
 
-        public sealed class LocalizableStringWithArguments : LocalizableString, IObjectWritable
+        public sealed class LocalizableStringWithArguments : LocalizableString
         {
             private readonly LocalizableString _messageFormat;
             private readonly string[] _formatArguments;
 
-            static LocalizableStringWithArguments()
-                => ObjectBinder.RegisterTypeReader(typeof(LocalizableStringWithArguments), reader => new LocalizableStringWithArguments(reader));
```

Comment:
ObjectBinder.RegisterTypeReader was how types specified how they could be deseriealized.  But it worked by registering a callback function at an index in an array, then emitting that array index into the serialized stream of bytes.  this, of course, coudl not work across processes or across sessions as that index woudl never be the same.  this is why this functionality is so practically useless.